core/src/main/java/hudson/security/ACL.java

```diff
@@ -77,6 +79,26 @@ public final void checkPermission(@Nonnull Permission p) {
         }
     }
 
+    public final void checkAnyPermission(@Nonnull Permission... permissions) {
+        boolean failed = !hasAnyPermission(permissions);
+
+        Authentication authentication = Jenkins.getAuthentication();
+        if (failed) {
+            String permissionsDisplayName = Arrays.stream(permissions)
```

Comment:
if an empty set of permissions was passed in does this work as expected?  (`permissionsDisplayName` would be an empty string) - so would the error message be correct?

Reply:
this method now throws an exception in that case